hedera-node/hedera-mono-service/src/main/java/com/hedera/services/contracts/execution/EvmTxProcessor.java

```diff
@@ -239,11 +177,14 @@ protected TransactionProcessingResult execute(
                     chargingResult.sender().incrementBalance(refundedWei);
                 }
             }
-            sendToCoinbase(coinbase, gasLimit - refunded, gasPrice, updater);
+            sendToCoinbase(
+                    coinbase, gasLimit - refunded, gasPrice, (HederaWorldState.Updater) updater);
             initialFrame.getSelfDestructs().forEach(updater::deleteAccount);
 
-            if (dynamicProperties.enabledSidecars().contains(SidecarType.CONTRACT_STATE_CHANGE)) {
-                stateChanges = updater.getFinalStateChanges();
+            if (((GlobalDynamicProperties) dynamicProperties)
```

Comment:
nit: Extract this check with something like
```
private boolean isSideCarTypeEnabled(SidecarType sidecarType) {
        return ((GlobalDynamicProperties) dynamicProperties)
                .enabledSidecars()
                .contains(sidecarType);
    }
```

Reply:
DONE